functorch/csrc/BatchRulesViews.cpp

```diff
@@ -272,6 +272,17 @@ std::tuple<Tensor, optional<int64_t>> select_batching_rule(const Tensor& self, o
   return std::make_tuple(result, 0);
 }
 
+std::tuple<Tensor, optional<int64_t>> _reshape_alias_batch_rule(const Tensor& self, optional<int64_t> bdim, const IntArrayRef shape, const IntArrayRef strides) {
+  (void) strides;
+  if (!bdim) {
+    return std::make_tuple(self.reshape(shape), nullopt);
+  }
+
+  auto new_shape = shape.vec();
+  new_shape.insert(new_shape.cbegin() + bdim.value(), self.size(bdim.value()));
+  return std::make_tuple(self.reshape(new_shape), bdim);
+}
```

Comment:
This algorithm looks correct to me. Did you have concerns about it, @vfdev-5?

Reply:
Well, I was not sure if this covers all cases. I also saw `_unsafe_view_batch_rule` which does almost the same thing we would like with this `_reshape_alias`:
https://github.com/facebookresearch/functorch/blob/43889f5e8ec22f1b5de914232aa9284911adb1fb/functorch/csrc/BatchRulesViews.cpp#L133-L141

Reply:
I think this should cover all the cases, we have tests for that :) (try to run `pytest test/ -v -k "reshape"`). 

Re: unsafe_view: `_unsafe_view` is actually the name for the operator. `_unsafe_view_batch_rule` is not saying that the batching rule is unsafe, it is saying that it is the batching rule for the "`unsafe_view`" operator